### PR TITLE
 refactor(compute): create Spot samples

### PR DESCRIPTION
**Pull request by ssvir**
## Description
implemented samples : compute_spot_check, compute_spot_create
Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist

- [ ] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] **Tests** pass:   `mvn clean verify` **required**
- [ ] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: compute/cloud-client/src/main/java/compute/spots/CreateSpot.java [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+    try (ImagesClient imagesClient = ImagesClient.create()) {
+      image = imagesClient.getFromFamily("debian-cloud", "debian-11").getSelfLink();
+    }
+    AttachedDisk attachedDisk = buildAttachedDisk(image, zone);
+    String machineTypes = String.format("zones/%s/machineTypes/%s", zone, "n1-standard-1");
+
+    // Send an instance creation request to the Compute Engine API and wait for it to complete.
+    return createInstance(projectId, zone, instanceName, attachedDisk, true,
+            machineTypes, null, null, false,
+            null, new ArrayList<>(), false, "STOP", null, false);
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/spots/CreateSpot.java:67]: Looks like most params are null or empty. Remove all non-essential parameters. Substitute their value inline in the code.

File: compute/cloud-client/src/main/java/compute/spots/CreateSpot.java [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+  //     be attached to the new instance.
+  // preemptible: boolean value indicating if the new instance should be preemptible
+  //     or not. Preemptible VMs have been deprecated and you should now use Spot VMs.
+  // spot: boolean value indicating if the new instance should be a Spot VM or not.
+  // instance_termination_action: What action should be taken once a Spot VM is terminated.
+  //     Possible values: "STOP", "DELETE"
+  // custom_hostname: Custom hostname of the new VM instance.
+  //     Custom hostnames must conform to RFC 1035 requirements for valid hostnames.
+  // delete_protection: boolean value indicating if the new virtual machine should be
+  //     protected against deletion or not.
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/spots/CreateSpot.java:99]: Remove all comments for non-essentials params.

File: compute/cloud-client/src/main/java/compute/spots/CreateSpot.java [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+  }
+
+  private static Instance buildInstanceResource(String instanceName, AttachedDisk disk,
+                                                String machineType, boolean externalAccess,
+                                                String externalIpv4, String subnetworkLink,
+                                                String internalIp,
+                                                List<AcceleratorConfig> acceleratorConfigs,
+                                                boolean preemptible, boolean isSpot,
+                                                String instanceTerminationAction,
+                                                String customHostName, boolean deleteProtection) {
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/spots/CreateSpot.java:133]: Same here. Remove all non-essential params and substitute their value inline.

File: compute/cloud-client/src/test/java/compute/spots/SpotsIT.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+import org.junit.Test;
+import org.junit.jupiter.api.Timeout;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.junit.runners.MethodSorters;
+
+@RunWith(JUnit4.class)
+@Timeout(value = 10, unit = TimeUnit.MINUTES)
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class SpotsIT {
```
Comment (Sita04) [inline compute/cloud-client/src/test/java/compute/spots/SpotsIT.java:41]:
```suggestion
public class SpotVmIT {
```

File: compute/cloud-client/src/test/java/compute/spots/SpotsIT.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  private static final String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
+  private static final String ZONE = "us-central1-b";
+  private static String INSTANCE_NAME;
+
+  // Check if the required environment variables are set.
+  public static void requireEnvVar(String envVarName) {
+    assertWithMessage(String.format("Missing environment variable '%s' ", envVarName))
+            .that(System.getenv(envVarName)).isNotEmpty();
+  }
+
```
Comment (Sita04) [inline compute/cloud-client/src/test/java/compute/spots/SpotsIT.java:51]: Use `Before` and `After` annotations to cleanup sysout between each test.

**Review by Sita04 — CHANGES_REQUESTED** [review]

File: compute/cloud-client/src/main/java/compute/spots/CreateSpot.java [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+    }
+
+    return build.build();
+  }
+
+  private static AttachedDisk buildAttachedDisk(String sourceImage, String zone) {
+    AttachedDiskInitializeParams initializeParams = AttachedDiskInitializeParams.newBuilder()
+            .setSourceImage(sourceImage)
+            .setDiskSizeGb(10)
+            .setDiskType(String.format("zones/%s/diskTypes/pd-standard", zone))
```
[the remark this answers is not in the document]
Comment (ssvir) [reply compute/cloud-client/src/main/java/compute/spots/CreateSpot.java:197]: I can't find it; all examples use String.format

File: compute/cloud-client/src/test/java/compute/spots/SpotsIT.java (hunk 5 above)
Comment (ssvir) [reply compute/cloud-client/src/test/java/compute/spots/SpotsIT.java:51]: I didn't use sysout in tests, so I don't see any reason to implement this staff

File: compute/cloud-client/src/main/java/compute/spots/CreateSpotVm.java [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests.
+    try (ImagesClient imagesClient = ImagesClient.create()) {
+      image = imagesClient.getFromFamily("debian-cloud", "debian-11").getSelfLink();
+    }
+    AttachedDisk attachedDisk = buildAttachedDisk(image, zone);
+    String machineTypes = String.format("zones/%s/machineTypes/%s", zone, "n1-standard-1");
+
+    // Send an instance creation request to the Compute Engine API and wait for it to complete.
+    return createInstance(projectId, zone, instanceName, attachedDisk, true, machineTypes, false);
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/spots/CreateSpotVm.java:64]: Print the response or a message here or in the main method as a signal for completion.

**Review by Sita04 — APPROVED** [review]
Kindly address the comment before merging. Thanks.

**Pull request merged**